packages/core/src/indexing/local-index-api.ts

```diff
@@ -149,9 +164,17 @@ export class LocalIndexApi implements IndexApi {
       await this.databaseIndexApi?.getPreviouslyIndexedModelsFromDatabase()
 
     const indexModelsArgs = []
-    for (const modelStreamId of models) {
+    for (const entry of models.entries()) {
+      const modelStreamId = entry[0]
+      const modelIndices = entry[1]
       this.logger.imp(`Starting indexing for Model ${modelStreamId.toString()}`)
-      const indexModelArgs = await _getIndexModelArgs(modelStreamId, this.repository)
+      let indexModelArgs = await _getIndexModelArgs(modelStreamId, this.repository)
+      if (modelIndices) {
```

Comment:
can we push this down into `_getIndexModelArgs` so all the logic for building the args is contained?